packages/react-aria-components/test/Toast.test.js

```diff
@@ -0,0 +1,290 @@
+  });
+
+  it('should move focus from the last toast to remaining toast when a the last toast is closed', async () => {
+    let {getAllByRole, getByRole, queryByRole} = render(<Example />);
+    let button = getByRole('button');
+
+    await user.click(button);
+    await user.click(button);
+
+    let toast = getAllByRole('alertdialog')[1];
+    let closeButton = within(toast).getByRole('button');
+    await user.click(closeButton);
+
+    toast = getByRole('alertdialog');
+    expect(document.activeElement).toBe(toast);
+    expect(toast).toHaveAttribute('data-focused', 'true');
+
+    closeButton = within(toast).getByRole('button');
+    await user.click(closeButton);
+
+    expect(queryByRole('alertdialog')).toBeNull();
+    expect(document.activeElement).toBe(button);
+  });
+
+  it('should support programmatically closing toasts', async () => {
+    const queue = new ToastQueue();
+    function ToastToggle() {
+      let [key, setKey] = React.useState(null);
+
+      return (
+        <>
+          <ToastRegion queue={queue}>
+            {({toast}) => (
+              <Toast toast={toast}>
+                <ToastContent>
+                  <Text slot="title">{toast.content}</Text>
+                </ToastContent>
+                <Button slot="close">x</Button>
+              </Toast>
+            )}
+          </ToastRegion>
+          <Button
+            onPress={() => {
+              if (key) {
+                setKey(null);
+                queue.close(key);
+              } else {
+                let key = queue.add('Toast is done!');
+                setKey(key);
+              }
+            }}>
+            {close ? 'Hide' : 'Show'} Toast
+          </Button>
+        </>
+      );
+    }
+
+    let {getByRole, queryByRole} = render(<ToastToggle />);
+    let button = getByRole('button');
+
+    await user.click(button);
+
+    act(() => jest.advanceTimersByTime(100));
+    let toast = getByRole('alertdialog');
+    let alert = within(toast).getByRole('alert');
+    expect(toast).toBeVisible();
+    expect(alert).toBeVisible();
+
+    await user.click(button);
+    expect(queryByRole('alertdialog')).toBeNull();
+    expect(queryByRole('alert')).toBeNull();
+  });
+
+  it('should support custom aria-label', async () => {
+    let queue = new ToastQueue();
+    let {getByRole} = render(
+      <>
+        <ToastRegion queue={queue} aria-label="Toasts">
+          {({toast}) => (
+            <Toast toast={toast}>
+              <ToastContent>
+                <Text slot="title">{toast.content}</Text>
+              </ToastContent>
+              <Button slot="close">x</Button>
+            </Toast>
+          )}
+        </ToastRegion>
+        <Button onPress={() => queue.add('Toast')}>Toast</Button>
+      </>
+    );
+
+    let button = getByRole('button');
+    await user.click(button);
+
+    let region = getByRole('region');
+    expect(region).toHaveAttribute('aria-label', 'Toasts');
+  });
+});
```

Comment:
Many of these tests were copied from RSP. In the future we could look at extracting them into a shared test util